Clover/app/src/main/java/org/floens/chan/core/presenter/BoardSetupPresenter.java

```
@@ -315,4 +347,120 @@ public long getId() {
             return code.hashCode();
         }
     }
+
+    private static class BoardSuggestionComparator implements Comparator<BoardSuggestion> {
+        public static final int SORT_BY_BOARD_CODE = 1 << 0;
+        public static final int SORT_BY_BOARD_NAME = 1 << 1;
+        public static final int SORT_ASCENDING = 1 << 8;
+        public static final int SORT_DESCENDING = 1 << 9;
+
+        private int options;
+
+        public BoardSuggestionComparator(BoardSortingInfo boardSortingInfo) {
+            options = boardSortingInfo.getSortingMode().value | boardSortingInfo.getSortingOrder().value;
+        }
+
+        @Override
+        public int compare(BoardSuggestion o1, BoardSuggestion o2) {
+            if (o1 == null && o2 == null) {
```

Probably should just throw an exception when either of parameters is null.